src/meta/src/controller/streaming_job.rs

```diff
@@ -1014,57 +1000,65 @@ impl CatalogController {
         Ok(version)
     }
 
+    /// TODO: make it general for other streaming jobs.
+    /// Currently only for replacing table.
     pub async fn finish_replace_streaming_job_inner(
         tmp_id: ObjectId,
         merge_updates: Vec<PbMergeUpdate>,
         table_col_index_mapping: Option<ColIndexMapping>,
-        creating_sink_id: Option<SinkId>,
-        dropping_sink_id: Option<SinkId>,
-        updated_sink_catalogs: Vec<SinkId>,
+        SinkIntoTableContext {
+            creating_sink_id,
+            dropping_sink_id,
+            updated_sink_catalogs,
+        }: SinkIntoTableContext,
         txn: &DatabaseTransaction,
         streaming_job: StreamingJob,
     ) -> MetaResult<(Vec<Relation>, Vec<PbFragmentWorkerSlotMapping>)> {
-        // Question: The source catalog should be remain unchanged?
-        let StreamingJob::Table(_, table, ..) = streaming_job else {
-            unreachable!("unexpected job: {streaming_job:?}")
-        };
+        let original_job_id = streaming_job.id() as ObjectId;
 
-        let job_id = table.id as ObjectId;
+        match streaming_job {
+            StreamingJob::Table(_source, table, _table_job_type) => {
```

Comment:
Shall we `let .. else { .. }` to avoid indent change?

Reply:
will add other branch soon...